File: Week 2/Problem Set 2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

//Prototypes
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //Get the text with user
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    //Coleman-Liau index calculate
    float L = (100 * (float) letters) / (float) words;
    float S = (100 * (float) sentences) / (float) words;

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    //Print the result
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}


//Count the letters of the text
int count_letters(string text)
{
    int cl = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //Set the count using only the uppercase or lowercase
        if (isupper(text[i]) || islower(text[i]))
        {
            cl++;
        }
    }
    return cl;
}

//Count the words of the text
int count_words(string text)
{
    //Every sentence has one word more than spaces
    int cw = 1;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //Sum spaces in the sentence
        if ((text[i] - ' ') == 0)
        {
            cw++;
        }
    }
    return cw;
}

//Count the sentences of the text
int count_sentences(string text)
{
    int cs = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //Sum <. ! ?> in the text
        if ((text[i] - '.') == 0 || (text[i] - '?') == 0 || (text[i] - '!') == 0)
        {
            cs++;
        }
    }
    return cs;
}

File: Week 6/Problem Set 6/sentimental-readability/readability.py
from cs50 import get_string


# Implements main function
def main():

    # Get the text
    text = get_string("Text: ")

    # Implements text parsing
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Coleman-Liau index calculate
    L = (100 * letters) / words
    S = (100 * sentences) / words

    index = 0.0588 * L - 0.296 * S - 15.8

    # Print result
    if index > 15:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index:.0f}")
# ==================================


# Count the letters of the text
def count_letters(text):
    cl = 0
    for i in range(len(text)):
        if text[i].isupper() or text[i].islower():
            cl += 1
    # print(cl)
    return cl
# ==================================


# Count the words of the text
def count_words(text):
    cw = 1
    for i in range(len(text)):
        if(text[i] == " "):
            cw += 1
    # print(cw)
    return cw
# ==================================


# Count the sentences of the text
def count_sentences(text):
    cs = 0
    for i in range(len(text)):
        if text[i] == "." or text[i] == "?" or text[i] == "!":
            cs += 1
    # print(cs)
    return cs
# ==================================


# Call main function
main()